### DISCUSSION TITLE
Feature Request: 'Copy To Person' option for Document Templates

### DISCUSSION THREAD

**Discussion by gitcodehelp** [sheltermanager/asm3#1499] [Q&A / Help]
We are a small shelter transitioning to ASM for its features and flexibilty.  It would be of great value to us if the 'Settings > Document Templates' table had a 'Copy To Person' column where we could select BroughtInBy, OriginalOwner, ReturnedBy, Owner, etc (or None).

Why?  Our shelter generates legal contracts for all intakes, including returns (see updated post #1493). Docs generated via:

Animal > Movements > (select adoption movement) > Document	

are copied/linked by default to the Media tab of the Owner person.  This is great for *Adoption* docs, but is undesirable for *Returns* (which are sometimes strays) -- those should be copied to ReturnedBy's Media tab.  Otherwise, staff must remember to do:

Media > Move/Copy > Move to a Person > from Owner to ReturnedBy

and must be especially careful *NOT* to do:

Media > Send PDF

from *Owner's* media tab, as the return doc will be emailed to *Owner*, not *ReturnedBy*.

Of course, you can initiate the Send PDF from the copy under Animal's Media, but then you have to manually fetch and type in (or copy/paste) the 'To:' address.  This is ok for the occasional Return intake, but is cumbersome for everyday Entry intakes, especially when clients would like their signed copy emailed to them.  Auto-copying docs to a target person would allow staff to do the 'Media > Send PDF' from the person where the "To:" email address will auto-populate. 

This feature would also greatly improve litter intakes for us, as we prefer to generate a separate contract doc for each Animal ID in the litter.  From the person's Media, all the individual litter intake contracts could then be selected at once as a group for signing and emailing.

Yes, for litter of say 6 kittens, some alternatives would be:
1) Create a single contract for the litter under the 1st kitten, sign it, then copy it to the person and the other kittens (too much copying, and staff/wordkeys do not know which 6 Animal IDs to put on the contract...)

2) Create the first kitten and its signed contract, clone it 5 times, then go to each clone and edit the signed doc (yes, that is possible last I checked...) to change Animal ID to the clone's number (again, too much work...)
   
Thanks again for your efforts!

**Comment by bobintetley** [comment]
Hmmm, that is tricky. I'm surprised this hasn't come up before. I can only assume it's because of your requirement to generate a special document for every person bringing an animal in. 

What sort of info is in this document? Might it be better implemented as an online form rather than a document template? Online forms will go straight to the media tab of the record of the person who completed the form, and they allow for one or more electronic signature fields. They also allow you to create a person record from the info supplied by the person if you don't already have a record for them. You can also opt to automatically send them a copy of the form they filled in.
